### DISCUSSION TITLE
Image (Pro) field with Polymorphic relation

### DISCUSSION THREAD

**Discussion by vdeep** [Laravel-Backpack/community-forum#704] [Q&A (Help)] [answered]
I'm having an issue with the Image (pro) field when using it to upload image to a model related via a polymorphic relation.

I've created an example site with minimum code to reproduce the issue:
https://github.com/vdeep/backpack-image-example

The site has three models:
1. Attachment
2. Author
3. Book

Book and Author have a polymorphic relation to the Attachment model.

When using the Image for 'Author' with name `attachment.filename`, it shows this error:

![image](https://github.com/Laravel-Backpack/community-forum/assets/2727830/098f6649-9f21-4f0e-a893-80684d40456d)

Is there any specific configuration to get this working? Or is there something I'm doing wrong?

**Comment by promatik** [comment, marked as answer]
Hi @vdeep!

In order to achieve this, you should use a "MorphOne (1-1 polymorphic) + subfields" field.
You have an [example in our demo](https://demo.backpackforlaravel.com/admin/monster/140/edit#relationship) 👌

Basically, you should setup your `attachment` relationship field with subfields, and one of those subfields (in your case the only one) should be your `filename` which is an `upload` field type.

Backpack will handle the rest for you 👌

```php
CRUD::addField([
    'name' => 'attachment',
    'label' => 'Attachment',
    'subfields' => [
        [
            'name' => 'filename',
            'type' => 'upload',
            'withFiles' => [
                'disk' => 'public',
                'path' => 'uploads',
            ],
        ],
		// more fields may go here...
    ],
]);
```

![image](https://github.com/Laravel-Backpack/community-forum/assets/1838187/abd4e60c-5ea9-4f6f-a5c1-6a6ad1b0916a)

You may add other fields to your relationship if needed, like a displayable attachment name.

Thank you by the provided sample code, it was helpful to quickly setup and fix the issue 👌

**Reply by vdeep** [reply]
Thanks @promatik ! This is working as expected.
